src/wordcount.js

```diff
@@ -227,9 +228,14 @@ export default class WordCount extends Plugin {
 	_calculateWordsAndCharacters() {
 		const txt = modelElementToPlainText( this.editor.model.document.getRoot() );
 
-		this.characters = txt.length;
+		this.characters = txt.replace( /\n/g, '' ).length;
 
-		this.words = ( txt.match( /[_a-zA-Z0-9À-ž]+/gu ) || [] ).length;
+		const wordsMatch = featureDetection.isUnicodePropertySupported ?
+			txt.match( new RegExp( '[\\p{L}\\p{N}\\p{M}\\p{Pd}\\p{Pc}]+', 'gu' ) ) :
```

Comment:
Maybe we could add what those groups are? I almost got those categories right (4 of 5) but I was reading them when doing mentions. Others might not remember/know them too well ;)